misk/src/main/kotlin/misk/scope/ActionScope.kt

```diff
@@ -0,0 +1,127 @@
+package misk.scope
+
+import com.google.inject.Key
+import com.google.inject.Provider
+import java.util.concurrent.Callable
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.reflect.KCallable
+import kotlin.reflect.KFunction
+import kotlin.reflect.KParameter
+import kotlin.reflect.KType
+import kotlin.reflect.KTypeParameter
+import kotlin.reflect.KVisibility
+
+@Singleton
+class ActionScope @Inject internal constructor(
```

Comment:
Should we offer guidance that scoped values must be immutable values? Otherwise propagate is sketchy

Reply:
(and franklin’s Op is deliberately mutable and not thread safe)